lib/update.go

```diff
@@ -220,63 +219,49 @@ type ServiceStatus struct {
 	Metrics    map[string]interface{}
 }
 
-// ServiceStatus reports status of the cron daemon
+// ServiceStatus reports status of the update daemon
 func (m *UpdateMethods) ServiceStatus(in *bool, out *ServiceStatus) error {
-	return fmt.Errorf("not finished")
-}
-
-// UpdateServiceStart starts the update service
-func UpdateServiceStart(ctx context.Context, repoPath string, updateCfg *config.Update, opts []Option) error {
-	if updateCfg == nil {
-		updateCfg = config.DefaultUpdate()
-	}
-
-	cli := cron.HTTPClient{Addr: updateCfg.Address}
-	if err := cli.Ping(); err == nil {
-		return fmt.Errorf("service already running")
+	res, err := update.Status()
+	if err != nil {
+		return err
 	}
 
-	var jobStore, logStore cron.JobStore
-	switch updateCfg.Type {
-	case "fs":
-		jobStore = cron.NewFlatbufferJobStore(repoPath + "/cron_jobs.qfb")
-		logStore = cron.NewFlatbufferJobStore(repoPath + "/cron_logs.qfb")
-	case "mem":
-		jobStore = &cron.MemJobStore{}
-		logStore = &cron.MemJobStore{}
-	default:
-		return fmt.Errorf("unknown cron type: %s", updateCfg.Type)
+	*out = ServiceStatus{
+		Name: res,
 	}
+	return nil
+}
 
-	svc := cron.NewCron(jobStore, logStore, updateFactory)
-	log.Debug("starting update service")
-	go func() {
-		if err := svc.ServeHTTP(updateCfg.Address); err != nil {
-			log.Errorf("starting cron http server: %s", err)
-		}
-	}()
-
-	return svc.Start(ctx)
+// UpdateServiceStartParams configures startup
+type UpdateServiceStartParams struct {
+	Ctx       context.Context
+	Daemonize bool
+
+	// TODO (b5): I'm really not a fan of passing these configuration-derived
+	// bits as parameters. Ideally this would come from the underlying instance
+	// these are needed because lib.NewInstance creates a cron client
+	// that intereferes with the start service process. We're currently getting
+	// around this by avoiding calls to lib.NewInstance, or passing in resulting
+	// params when called. We should clean this up.
+	RepoPath  string
+	UpdateCfg *config.Update
 }
 
 // ServiceStart ensures the scheduler is running
-func (m *UpdateMethods) ServiceStart(in, out *bool) error {
-	local, ok := m.inst.cron.(*cron.Cron)
-	if !ok {
-		return fmt.Errorf("service already running")
+func (m *UpdateMethods) ServiceStart(p *UpdateServiceStartParams, started *bool) error {
+	// TODO (b5) - these work when the API is running
+	if p.RepoPath == "" && m.inst != nil {
+		p.RepoPath = m.inst.QriPath()
 	}
-
-	ctx := context.Background()
-	if err := local.Start(ctx); err != nil {
-		return err
+	if p.UpdateCfg == nil && m.inst != nil {
+		p.UpdateCfg = m.inst.Config().Update
 	}
-	return nil
+	return update.Start(p.Ctx, p.RepoPath, p.UpdateCfg, p.Daemonize)
```

Comment:
`started` is never assigned to. Instead, check that `Start` returns an error, and if nil, set `started` to true.